controllers/walletsController.js

```diff
@@ -22,6 +23,28 @@ const getUserWallet = async (req, res) => {
   }
 }
 
+/**
+ * Get the wallet details of user, if a username is provided
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+const getWallet = async (req, res) => {
+  try {
+    const { params: { username } = {} } = req
```

Comment:
I presume `user1` will be allowed to get the wallet data of only `user1`.
Here, `user1` can pass the username of `user2` and get their wallet details!

Thus, we should have validation if the username passed with path params corresponds to the cookie owner.
We can get the authenticated user's data using the object at `req.userData`.

Thus, the `getUserId` utility is not required here.

Reply:
We are authorizing this route for super users. So, it should not be a problem.